web/src/components/auth/SignUpModal.tsx

```diff
@@ -61,7 +62,14 @@ export const SignUpModal = (): ReactElement => {
             style="primary"
             className="width-100"
             onClick={() => {
-              analytics.event.guest_modal.click.go_to_myNJ_registration();
+              if (
+                userData?.preferences.returnToLink ===
+                `${ROUTES.dashboard}?${QUERIES.openTaxFilingsModal}=true`
+              ) {
+                analytics.event.myNJ_prompt_modal_complete_button.click.go_to_myNJ_registration();
+              } else {
+                analytics.event.guest_modal.click.go_to_myNJ_registration();
+              }
```

Comment:
typically when there's a significant amount of logic in an onClick (more than 2ish lines, or anything it has additional nesting like and if/else her) I recommend extracting and making it a named function, for readability